Orange/canvas/document/schemeedit.py

```diff
@@ -1029,6 +1047,30 @@ def changeEvent(self, event):
 
         QWidget.changeEvent(self, event)
 
+    def tryInsertNode(self, link, new_node_desc, pos):
+        source_node = link.source_node
+        sink_node = link.sink_node
+
+        if nodes_are_compatible(source_node.description, new_node_desc) and \
+                nodes_are_compatible(new_node_desc, sink_node.description):
```

Comment:
The test here should also be constrained on the existing `link.source_channel` and `link.sink_channel`.

E.g. If I use:

<img width="258" alt="screen shot 2018-08-02 at 11 21 41" src="https://user-images.githubusercontent.com/4716745/43575121-466a8148-9646-11e8-8bb0-efce9fb5dfa4.png">

and drop an *Impute* widget on the link, I get a
```
------------------- IncompatibleChannelTypeError Exception --------------------
Traceback (most recent call last):
  File "/Users/aleserjavec/workspace/orange3/Orange/canvas/document/schemeedit.py", line 1102, in eventFilter
    self.tryInsertNode(link, desc, pos)
  File "/Users/aleserjavec/workspace/orange3/Orange/canvas/document/schemeedit.py", line 1066, in tryInsertNode
    new_node, possible_links[0][0][1]),  # first link, first entry, output
  File "/Users/aleserjavec/workspace/orange3/Orange/canvas/scheme/link.py", line 125, in __init__
    % (source_channel.type, sink_channel.type)
Orange.canvas.scheme.errors.IncompatibleChannelTypeError: Cannot connect 'Orange.base.Model' to 'Orange.base.Learner'
-------------------------------------------------------------------------------
```
The set of possible links below should also be filtered correspondingly.

P.S.
The same also applies to the 'Insert widget` context menu action.